spec/models/user_spec.rb

```diff
@@ -0,0 +1,18 @@
+require 'rails_helper'
+
+RSpec.describe User, type: :model do
+  it 'is not valid without a name' do
+    user = User.new
+    expect(user).to_not be_valid
+  end
+
+  it 'is valid with a valid name' do
+    user = User.new(name: 'John Doe')
+    expect(user).to be_valid
+  end
+
+  it 'has a non-negative post counter' do
+    user = User.new(name: 'Jane Smith', post_counter: -1)
+    expect(user).to_not be_valid
+  end
+end
```

Comment:
Nice job so far. However, as part of the project requirements, any custom methods in the models should have a corresponding unit test. In view of this kindly add unit test for the `recent_posts` method in the user model here. :wink: